lib/loglevel.js

```diff
@@ -189,68 +195,108 @@
       self.methodFactory = factory || defaultMethodFactory;
 
       self.getLevel = function () {
-          if (userLevel != null) {
-            return userLevel;
-          } else if (defaultLevel != null) {
-            return defaultLevel;
-          } else {
-            return inheritedLevel;
+          if (userLevel !== null) {
+              return userLevel;
+          }
+          if (defaultLevel !== null) {
+              return defaultLevel;
           }
+          if (inheritedLevel === null) {
+              inheritedLevel = self.getInheritedLevel();
+          }
+          return inheritedLevel;
       };
 
-      self.setLevel = function (level, persist) {
-          userLevel = normalizeLevel(level);
-          if (persist !== false) {  // defaults to true
-              persistLevelIfPossible(userLevel);
+      self.getInheritedLevel = function() {
+          if( defaultLogger && self.parentKey ) {
+              return defaultLogger.getLogger(self.parentKey).getLevel();
           }
-
-          // NOTE: in v2, this should call rebuild(), which updates children.
-          return replaceLoggingMethods.call(self);
+          if( !defaultLogger || defaultLogger===self ) {
+              return normalizeLevel("WARN");
+          }
+          return normalizeLevel(
+              defaultLogger ? defaultLogger.getLevel() : "WARN" );
```

Comment:
I think this method can be simplified: `defaultLogger` is always set, and with `parentKey` gone that bit becomes `defaultLogger.getLogger(...keyHierarchy.slice(0, -1)).getLevel()`, which should work for all loggers except the root so we don't need a parentKey check. We don't need normalizeLevel either. Could be something like:

```js
if (defaultLogger === self) {
  return self.levels.WARN;
} else {
  return defaultLogger.getLogger(...keyHierarchy.slice(0, -1)).getLevel();
}